test/registry.c

```diff
@@ -74,17 +75,35 @@ struct test_option_group {
     struct test_option options[10];
 };
 
+#define POPULARITY_NONE 0
+static const char *
+popularity_attr(enum rxkb_popularity popularity)
+{
+    switch ((int) popularity) {
```

Comment:
```suggestion
    switch ((int) popularity) {
```
is the `(int)` required here? technically this removes a little bit of type-safety and we have little enough of that in C as it is :)

Reply:
I love type-safety, but here I faced the issue that `POPULARITY_NONE` is not a member of the enum and raises a compiler warning. Well, the next option is to deal with it in the `default` case.

Reply:
hmm, I wonder if it makes sense to work around this by doing a 
```
enum popularity {
   POP_STANDARD = POPULARITY_STANDARD,
   POP_EXOTIC = POPULARITY_EXOTIC,
   POP_UNDEFINED = PROP_EXOTIC + 1
}
```
and using that. That would also allow you to return a `struct config_item` from `parse_config_item()` and override the popularity if `POP_UNDEFINED`. All without leaking the undefined value into the public API.

Then again, the public API was explicitly designed so we can use `0` as none case, so all this could be over the top :)

Reply:
I propose the following:
- Use the idea of the new enum in test. We ensure we keep up-to-date with assertions.
- Using an extended enum in `struct config_item` would require to also check we keep up-to-date, then add a test at _each_ call of `parse_config_item`. I think we can do simpler: just initialize all members of `struct config_item` in the callers of `parse_config_item`, and not just popularity as previously.

I updated the code to implement this. What do you think?